test/test_type_hints.py

```diff
@@ -170,6 +170,20 @@ def test_doc_examples(self):
             except subprocess.CalledProcessError as e:
                 raise AssertionError("mypy failed.  Look above this error for mypy's output.")
 
+    def test_type_hint_examples(self):
+        """
+        Runs mypy over all the test examples present in
+        type_hint_tests directory
+        """
+
+    examples = os.listdir(os.getcwd() + '/test/type_hint_tests/')
+    for i in examples:
+        process = subprocess.Popen(['mypy', os.getcwd()
+                                   + '/test/type_hint_tests/' + i],
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.PIPE)
+        (stdout, stderr) = process.communicate()
+        print(stdout)
```

Comment:
Right now this doesn't cause the test to fail if mypy errored out, right? We could copy the logic from https://github.com/pytorch/pytorch/blob/12fb8148e48d3e39d16832a84ac4b59182a00c4e/test/test_type_hints.py#L161-L171 that runs mypy and tries to see if there are any problems.

Reply:
Right now with the current changes, it throws Assertion Error
```
Traceback (most recent call last):
  File "test/test_type_hints.py", line 112, in <module>
    class TestTypeHints(TestCase):
  File "test/test_type_hints.py", line 189, in TestTypeHints
    raise AssertionError('Type hint tests not passed in ', i)
AssertionError: ('Type hint tests not passed in ', 'size.py')
```

Reply:
@zou3519  I am not sure how to use `torch.testing.` to fail the test, so that it prints like you'd expect rather than throwing Assertion error seperately. I tried `unittest.skip` but it is the same as Assertion Error.

Reply:
Also i don't know why lint test fails. On my machine flake8 check passes properly.

Reply:
AssertionError is fine, that's what the other test does

Reply:
Then what is the next step ? Should I fix the Size definition now finally?

Reply:
And regarding passing annotation in linting test, what do I add ?

Reply:
> Also i don't know why lint test fails. On my machine flake8 check passes properly.

I did:
```
cd test/type_hint_tests/
flake8
```
and it gave me the following output:
```
./size.py:9:1: W391 blank line at end of file
```
Maybe there is something different about your flake8 setup?

Reply:
I'm on Ubuntu and I run 
```
flake8 test/test_type_hints.py
```
It throws no error. The issue is with annotation, i guess as CI suggests. What changes do I make there ?

Reply:
> It throws no error. The issue is with annotation, i guess as CI suggests. What changes do I make there ?

`size.py` has an extra line at the end of the file. Deleting it should make the flake8 warning go away. In the long run it would be nice to figure out what is up with your flake8, but I am not an expert with that.

> Then what is the next step ? Should I fix the Size definition now finally?

Yes, that sounds great

Reply:
My flake8 error now says
`
test/test_type_hints.py:199:16: W292 no newline at end of file
`
I definitely have to check flake8.

Reply:
Yeah, it sounds like test/test_type_hints.py needs a newline at the end.

Reply:
I removed the line to remove the flake8 error as you suggested. It's required.